packages/react-components/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,26 @@
+import {
+  Switch as ReactAriaSwitch,
+  SwitchProps as ReactAriaSwitchProps,
+} from "react-aria-components";
+
+import "./Switch.css";
+
+export interface SwitchProps extends ReactAriaSwitchProps {
+  children?: React.ReactNode;
+  /* Label positioning relative to switch */
+  labelPosition?: "left" | "right";
+}
+
+export default function Switch({
+  labelPosition = "right",
+  children,
+  ...props
+}: SwitchProps) {
+  return (
+    <ReactAriaSwitch className={`bcds-react-aria-Switch`} {...props}>
+      {labelPosition === "left" && <>{children}</>}
```

Comment:
@ty2k this pair of conditionals for positioning the label feels like a pretty inelegant solution, if you have a better way to do this would love to hear it!

Reply:
You could switch the flex direction of the thing like `flex-direction: row` and `flex-direction: row-reverse` based on that `labelPosition` prop, but this is probably the more clear solution. We can always complicate the heck out of this when we have to make all these components work for `rtl` instead of just `ltr` some day. :)